# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../spqmail/0002-qmail-smtpd-UCSPISSL.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 40594d8d2aff69de3c84356482b40b2c8b9a53ac Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Fri, 16 Nov 2012 16:42:51 +0100
Subject: [PATCH] qmail-smtpd: check if the environment variable UCSPITLS=1 to
 provide STARTTLS support

---
 qmail-smtpd.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/qmail-smtpd.c b/qmail-smtpd.c
index 0571a24..e4be563 100644
--- a/qmail-smtpd.c
+++ b/qmail-smtpd.c
@@ -1546,6 +1546,11 @@ void setup()
   relayclient = env_get("RELAYCLIENT");
   requireauth = env_get("REQUIREAUTH");
 
+  /* check if the environment variable UCSPITLS=1 to provide STARTTLS support*/
+  x = env_get("UCSPITLS");
+  if (x) { scan_ulong(x,&tls_available); }
+  if (tls_available < 1) tls_available=0;
+
   dohelo(remotehost);
 }
 
@@ -1574,8 +1579,6 @@ int main(int argc, char **argv)
   setup();
   p0f_query();
 
-  tls_available = !!env_get("UCSPITLS");
-
   if (ipme_init() != 1) die_ipme();
   while (1) {
     i+=strlen(greeting.s+i)+1;
-- 
1.7.10.2


From e1f4d656995ec4425403d938a6de750164877df1 Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Fri, 16 Nov 2012 16:52:02 +0100
Subject: [PATCH] qmail-smtpd: changed to check if the environment variable
 UCSPISSL=1 which means that TLS is already started

---
 qmail-smtpd.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/qmail-smtpd.c b/qmail-smtpd.c
index e4be563..a21fb83 100644
--- a/qmail-smtpd.c
+++ b/qmail-smtpd.c
@@ -1551,6 +1551,11 @@ void setup()
   if (x) { scan_ulong(x,&tls_available); }
   if (tls_available < 1) tls_available=0;
 
+  /* if the environment variable UCSPISSL=1 TLS is already started */
+  x = env_get("UCSPISSL");
+  if (x) { scan_ulong(x,&tls_started); }
+  if (tls_started < 1) tls_started=0;
+
   dohelo(remotehost);
 }
 
-- 
1.7.10.2

